rollbar-java/src/main/java/com/rollbar/notifier/Rollbar.java

```diff
@@ -1,4 +1,633 @@
+   */
+  public void log(Throwable error, Map<String, Object> custom, String description) {
+    log(error, custom, description, null);
+  }
+
+  /**
+   * Record a debugging message at the level returned by {@link Rollbar#level} (WARNING unless level
+   * is overridden).
+   *
+   * @param message the message.
+   */
+  public void log(String message) {
+    log(null, null, message, null);
+  }
+
+  /**
+   * Record a message with extra information attached at the default level returned by {@link
+   * Rollbar#level}, (WARNING unless level overridden).
+   *
+   * @param message the message.
+   * @param custom the extra information.
+   */
+  public void log(String message, Map<String, Object> custom) {
+    log(null, custom, message, null);
+  }
+
+  /**
+   * Record a message at the level specified.
+   *
+   * @param message the message.
+   * @param level the level.
+   */
+  public void log(String message, Level level) {
+    log(null, null, message, level);
+  }
+
+  /**
+   * Record a message with extra information attached at the specified level.
+   *
+   * @param message the message.
+   * @param custom the extra information.
+   * @param level the level.
+   */
+  public void log(String message, Map<String, Object> custom, Level level) {
+    log(null, custom, message, level);
+  }
+
+  /**
+   * Record an error or message with extra data at the level specified. At least ene of `error` or
+   * `description` must be non-null. If error is null, `description` will be sent as a message. If
+   * error is non-null, description will be sent as the description of the error. Custom data will
+   * be attached to message if the error is null. Custom data will extend whatever {@link
+   * Config#custom} returns.
+   *
+   * @param error the error (if any).
+   * @param custom the custom data (if any).
+   * @param description the description of the error, or the message to send.
+   * @param level the level to send it at.
+   */
+  public void log(Throwable error, Map<String, Object> custom, String description, Level level) {
+    process(error, custom, description, level);
+  }
+
+  /**
+   * Get the level of the error or message. By default: CRITICAL for {@link Error}, ERROR for other
+   * {@link Throwable}, WARNING for messages. Override to change this default.
+   *
+   * @param error the error.
+   * @return the level.
+   */
+  public Level level(Throwable error) {
+    if (error == null) {
+      return Level.WARNING;
+    }
+    if (error instanceof Error) {
+      return Level.CRITICAL;
+    }
+    return Level.ERROR;
+  }
+
+  private void process(Throwable error, Map<String, Object> custom, String description,
+      Level level) {
+
+    // Pre filter
+    if (config.filter() != null && config.filter().preProccess(level, error, custom, description)) {
+      return;
+    }
+
+    // Gather information to build a payload.
+    Data data = buildData(error, custom, description, level);
```

Comment:
It took me a second to figure out why this returns the data instead of the builder, and it is because the later transforms take in the whole data object in order to work.